### PR TITLE
remove dependency on maven-shared-io

### PR DESCRIPTION
**Pull request by elharo**
@rfscholte @ian-lavallee

### CODE REVIEW

File: src/main/java/org/apache/maven/plugins/assembly/io/ClasspathResourceLocatorStrategy.java [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+        Location location = null;
+
+        if ( resource != null )
+        {
+            location = new URLLocation( resource, locationSpecification, tempFilePrefix, tempFileSuffix,
+                                        tempFileDeleteOnExit );
+        }
+        else
+        {
+            messageHolder.addMessage( "Failed to resolve classpath resource: " + locationSpecification
```
Comment (ian-lavallee) [inline src/main/java/org/apache/maven/plugins/assembly/io/ClasspathResourceLocatorStrategy.java:60]: Why isn't this just an exception?

File: src/main/java/org/apache/maven/plugins/assembly/io/FileLocation.java [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+    {
+        if ( ( channel != null ) && channel.isOpen() )
+        {
+            try
+            {
+                channel.close();
+            }
+            catch ( IOException e )
+            {
+                //swallow it.
```
Comment (ian-lavallee) [inline src/main/java/org/apache/maven/plugins/assembly/io/FileLocation.java:72]: Why do we swallow the exception? Isn't this bad practice?

File: src/main/java/org/apache/maven/plugins/assembly/io/MessageLevels.java [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+     */
+    public static final int LEVEL_INFO = 1;
+    /**
+     * Warning.
+     */
+    public static final int LEVEL_WARNING = 2;
+    /**
+     * Error
+     */
+    public static final int LEVEL_ERROR = 3;
```
Comment (ian-lavallee) [inline src/main/java/org/apache/maven/plugins/assembly/io/MessageLevels.java:48]: Why not use enums here?

File: src/main/java/org/apache/maven/plugins/assembly/io/ClasspathResourceLocatorStrategy.java (hunk 1 above)
Comment (elharo) [reply src/main/java/org/apache/maven/plugins/assembly/io/ClasspathResourceLocatorStrategy.java:60]: perhaps it should be. For now this just tries to maintain existing behavior.

File: src/main/java/org/apache/maven/plugins/assembly/io/FileLocation.java (hunk 2 above)
Comment (elharo) [reply src/main/java/org/apache/maven/plugins/assembly/io/FileLocation.java:72]: In general, yes. Exceptions are often swallowed when closing a resource though, under the assumption that nothing else important will happen with it.

Also, note that most of this code is just moved over from maven-shared-io. It's not actually new. I just wanted eliminate the dependencies on the much larger parts of maven-sahred-io we're not using here.

File: src/main/java/org/apache/maven/plugins/assembly/io/MessageLevels.java (hunk 3 above)
Comment (elharo) [reply src/main/java/org/apache/maven/plugins/assembly/io/MessageLevels.java:48]: I suspect this code predates Java 5.

**Pull request merged**